### ISSUE TITLE
Support SSL certificate rotation

### ISSUE THREAD

**Issue by robobario** [kroxylicious/kroxylicious#104]
With https://github.com/kroxylicious/kroxylicious/pull/103 we will support TLS/SSL on the proxy port

We should in theory be able to change the certificates without dropping connections or restarting:

https://goldius.medium.com/netty-reloading-ssl-tls-certificate-2078cdf6bfc1

**Comment by k-wall** [comment]
Agreed, for the multi-tenant use case, eventually we'd want the ability to:

- dynamically add/remove tenants
- allow a per-certificate TLS certificate (exploiting Netty's SniHandler/DomainWildcardMapping)
- allow TLS certificates to be rolled before they expired

all without requiring a kroxy restart.

**Comment by Hakky54** [comment]
This also might be usefull [SSL reloading with netty](https://github.com/Hakky54/java-tutorials/tree/main/instant-server-ssl-reloading-with-netty/netty-server). In the past I did something similar and made it available there as a tutorial. Hope it will be useful

**Comment by k-wall** [comment]
Not a priority for 0.5.0

**Comment by k-wall** [comment]
This problem is wider than the serving certificate.  It also applies to TLS trust.   Kroxylicious ought to be be prepared to reload that too.
